pkg/kubelet/images/image_manager.go

```diff
@@ -136,12 +186,31 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, pod *v1.Pod, conta
 	}
 
 	present := imageRef != ""
-	if !shouldPullImage(container, present) {
+
+	var pulledBySecret, ensuredBySecret bool
+	if present && utilfeature.DefaultFeatureGate.Enabled(features.KubeletEnsureSecretPulledImages) {
+		pulledBySecret, ensuredBySecret, err := m.isEnsuredBySecret(imageRef, spec, pullSecrets)
```

Comment:
this looks like we're shadowing pulledBySecret and ensuredBySecret. Shouldn't this be `=`?

Reply:
nod looks like a regression after adding the earlier feature gate check...

Reply:
Fixed.